erpnext/accounts/doctype/pos_invoice/pos_invoice.py

```diff
@@ -139,7 +139,12 @@ def validate_stock_availablility(self):
 				if allow_negative_stock:
 					return
 
-				available_stock = get_stock_availability(d.item_code, d.warehouse)
+				if frappe.db.get_value('Item', d.item_code, 'is_stock_item'):
```

Comment:
Let's handle these check inside the `get_stock_availablity()` since the same method is being used in `point_of_sale.py`

Reply:
Done.